src/js/oncoprintruleset.js

```diff
@@ -935,11 +941,69 @@ var Rule = (function () {
     return Rule;
 })();
 
+var GradientCategoricalRuleSet = (function() {
+
+        function GradientCategoricalRuleSet(params) {
+	        RuleSet.call(this, params);
+	        this.prototype = Object.create(RuleSet.prototype);
+                // For the GradientCategoricalRuleSet a datum must always have a 
+                // value (for heatmap clustering) and can have a category attribute. 
+                // A datum is 'NA' when not meeting the requirements for the GradientRuleSet.
+                // To achieve correct evaluation, the CategoricalRuleSet is asked not to
+                // contribute an `NA` rule.
+                this.gradientRuleSet = new GradientRuleSet(params);
+                this.categoricalRuleSet = new CategoricalRuleSet(params, omitNArule);
+        }
+
+        // inherit methods from RuleSet
+        GradientCategoricalRuleSet.prototype = Object.create(RuleSet.prototype);
+        Object.defineProperty(GradientCategoricalRuleSet.prototype, 'constructor', { 
```

Comment:
why are you doing this?

Reply:
Was mentioned in the [MDN manual](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Inheritance#Setting_Teacher()'s_prototype_and_constructor_reference) on function class inheritance in JS.

Reply:
but we don't do this anywhere else in the project and it hasn't seemed to cause any problems. For consistency let's leave it out

Reply:
I have removed this construct.